jigglypuffRL/deeprl/dqn/utils.py

```diff
@@ -0,0 +1,33 @@
+import torch
+import torch.nn as nn
+from torch.autograd import Variable
+import numpy as np
+import random
+
+class DQN_Mlp(nn.Module):
+    def __init__(self, env):
+        super(DQN_Mlp, self).__init__()
+        
+        self.env = env
+        
+        self.layers = nn.Sequential(
+            nn.Linear(self.env.observation_space.shape[0], 128),
+            nn.ReLU(),
+            nn.Linear(128, 128),
+            nn.ReLU(),
+            nn.Linear(128, self.env.action_space.n)
+        )
+        
+    def forward(self, x):
+        return self.layers(x)
+    
+    def act(self, state, epsilon):
+        if np.random.rand() > epsilon:
+            state   = Variable(torch.FloatTensor(state).unsqueeze(0), volatile=True)
+            q_value = self.forward(state)
+            action  = q_value.max(1)[1].data[0].item()
+        else:
+            action = random.randrange(self.env.action_space.n)
+        return action
```

Comment:
@ajaysub110 We might be able to use MLPCritic for DQNs right? Because afaik its a Value function MLP which is what DQN requires. If that's the case, this might be redundant.

Reply:
Yeah, had mentioned this in my review as well

Reply:
yea initially thought of using MLPCritic but then thought to keep classes for policy gradient and DQN different. Will change it to MLPCritic

Reply:
I still want this to be different from `MLPCritic` class as this might cause confusion to the user. Can we not keep policy gradient and DQN implementations of the network different?

Reply:
Critic in the actor-critic setting is supposed to be a value function-based method. DQN formulation is pretty much a value function based algorithm. So, I'd still suggest using the MLPCritic methods otherwise it becomes a bit useless as an independent class.

Reply:
I will be having some modifications in this class itself when I will be having Duelling DQN, this is why i wanted it to be a different class.

Reply:
Moreover won't we be taking the action based on the q-value in DQN? I did not understand as to what should I pass as action in the MLPCritic method.

Reply:
The Critic function will take a state-action pair while DQN will only take state to return different values of actions. Also, theoretically doesn't the Critic learn the advantage values while the Q function learn Q-values?They can't be made the same. Also, the value_hist should then become q_hist.

Reply:
You can use `MLPValue` from `Values.py`